app/src/main/java/app/passwordstore/ui/crypto/PasswordDialog.kt

```diff
@@ -21,15 +21,27 @@ import app.passwordstore.util.extensions.unsafeLazy
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
 
 /** [DialogFragment] to request a password from the user and forward it along. */
-class PasswordDialog : DialogFragment() {
+class PasswordDialog(private val enabledCache: Boolean, private val clearCache: Boolean) :
```

Comment:
Adding constructor parameters here will cause issues during configuration changes when the framework tries to find a no-args constructor. You'll need to use this [newInstance pattern](https://github.com/android-password-store/Android-Password-Store/blob/78357a5d4943014edb5047d5b88ec66f9281584e/app/src/main/java/app/passwordstore/ui/dialogs/FolderCreationDialogFragment.kt#L70) to pass in the booleans in a Bundle.

Reply:
Thank you, I'll try to figure out what needs to be done.

Reply:
Ok, I think I got it right. PasswordDialog is now instantiated via newInstance, passing arguments as Bundle.